fsspec/caching.py

```diff
@@ -766,15 +766,45 @@ def _read_cache(self, start, end, start_block_number, end_block_number):
             return b"".join(out)
 
 
-caches = {
-    "none": BaseCache,
-    None: BaseCache,
-    "mmap": MMapCache,
-    "bytes": BytesCache,
-    "readahead": ReadAheadCache,
-    "block": BlockCache,
-    "first": FirstChunkCache,
-    "all": AllBytes,
-    "parts": KnownPartsOfAFile,
-    "background": BackgroundBlockCache,
-}
+caches = {}
+
+
+def add_cache(cls, overload=False):
```

Comment:
We use "overwrite" or "clobber" in other places.

Reply:
```
❯ git grep overwrite=
fsspec/implementations/dbfs.py:    def _create_handle(self, path, overwrite=True):

❯ git grep clobber=
docs/source/changelog.rst:- Set default clobber=True on impl register
fsspec/registry.py:def register_implementation(name, cls, clobber=True, errtxt=None):
```

so I will for consistency
- name it `register_cache` 
- change to `clobber=True` for consistency, but I think it would be better to default to `False` in both cases. WDYT?

Reply:
pushed meanwhile all above and otherwise recommended changes

Reply:
For the DBFS, overwrite I think is the right choice. For register_implementation, I could be persuaded, but perhaps it shouldn't happen in this PR.